### ISSUE TITLE
RxJava should be declared as a api dependency 

### ISSUE THREAD

**Issue by StefMa** [beworker/knot#20]
Currently the dependency to rxjava is an `implementation` dependency.
That means it is **not available** for consumers of this library at **compile time**.
But you have open APIs to rxjava (like the `knot.state` property).
That means if a project wants to use `knot` and declare it as a dependency it will not work.
It will lead to compile time errors.

Long story short:
Rxjava should be declared as an api dependency so other project can directly start with knot without adding rxjava to their dependencies.

**Comment by beworker** [comment]
Thanks for noticing that. I've updated the "Downloads" section to make it more clear, that the Rx2 libraries are required at runtime.

It was done rather on purpose to decouple the library from a certain Rx version. Similar to how kotlin libraries are expected, the library expects Rx2 be present at runtime. This avoids statements like "Knot is still compiled for Rx2 version x.y.z". As long as Rx2 releases remain binary-compatible, Knot will just work.

**Comment by StefMa** [comment]
The one thing has nothing to do with the other.
Your library depend on RxJava. That is a fact. Regardless of the dependency 'type' (implementation or api), RxJava have to be available at least runtime so that your library work as expected.
You can do absolutely nothing to make sure that a new RxJava version will break your library. You have to always trust the RxJava authors to keep compatibility between versions.

Anyways with just mention it in your download section you will just make it "harder" for new consumers. Because they have to ass two dependencies (instead of just one, knot).

Gradle will always pick the highest version of a dependency in the whole dependency graoh, including transitive dependencies.
What does this mean for you?
Well, if you ship RxJava 2.1.0 (via an api dependency), but the consumer declare 2.1.5 which introduces a bug in some code you use, your library will break. Because, as said, gradle will pick 2.1.5 anyways.
It is the same behaviour if you shoot it as a implementation dependency. Nothing will change here!
The only differences is, and this is important, that newbies can directly start using knot with a single dependency if you declare RxJava as an api dependency. That is really the only difference.
But what is also more important than this. Your knit api share code from RxJava. Imagine someone wants to use your library, declare knot but forgot RxJava. What will happen? He can't compile(!) it's project. But, funny thing, it would work at runtime. Because RxJava is available at runtime, but not at compile time. A consumer would also be unhappy und doesn't know why it code doesn't compile and will probably throw out knot of it's code base.

In a nutshell:
The correct approach of declaring a dependency in Gradle is like:
If you share a api use api.
If not, so your dependency is hidden in library only, use implementation.

**Comment by beworker** [comment]
Fixed in 153966c31eb1a3e16d05e3d8d33c0f4a32948c1e. Thanks for caring!
